### PR TITLE
Adding Noahs changes & trained model to Florians / Omars CSD prediction repo

### PR DESCRIPTION
**Pull request by ayla-s**
Noah started using Florians model & data preparation code ~ mid 2024.
Since he only copied the parts he needed to his [BSc thesis GitLab repo](https://gitlab.lrz.de/compmass/students/charge_state_pred_eval/-/tree/main?ref_type=heads), this is now the integration of his work back into the dlomix repo.

#### His work evolved around:
* different data preparation (MS1-based CSD calculation, adjusted from Groshkov et al., 2024; code to be integrated elsewhere - ongoing)
* testing different loss functions
* comparing different SOTA CSD prediction models

His code regarding new model elements & model training, as well as the final pretrained model are added in this pull request.

### CODE REVIEW

**Review by omsh — APPROVED** [review]
Thanks a lot Ayla for your support and contributions.
